### ISSUE TITLE
what's the meaning of "scaling down gradients" in training log?

### ISSUE THREAD

**Issue by imaglib** [Russell91/nlpcaffe#4]
Hi Russell:
  During the training , there are the std log " Gradient clipping: scaling down gradients (L2 norm 0.243775 > 0.24) by scale factor 0.984513".  The 0.24 is the threshold of gradients? Why set this value , empirical value? RNN has drawback of gradient vanishing, Why there is bigger value ?

**Comment by Russell91** [comment]
Gradient clipping addresses the problem of exploding gradients, not vanishing gradients. The clipping implementation is also a feature of the core caffe codebase though, and is not unique to nlpcaffe. Most LSTM papers that I am aware of use a similar form of gradient clipping. You can adjust the threshold in the create_lm.py file and see if it gets you better quality though.

**Comment by imaglib** [comment]
I mean the RNN has drawback of gradient vanishing, so i think the gradient should be samller, why use the gradient clipling to limit the gradient?
